File: examples/lh-server/java/21-structdefs/src/main/java/io/littlehorse/docs/structdefs/Car.java
package io.littlehorse.docs.structdefs;

import io.littlehorse.sdk.worker.LHStructDef;
import io.littlehorse.sdk.worker.LHStructField;

@LHStructDef("car")
public class Car {

    @LHStructField(description = "The vehicle manufacturer.")
    private String make;

    @LHStructField(description = "The manufacturer model name.")
    private String model;

    @LHStructField(description = "The model year.")
    private int year;

    public Car() {}

    public Car(String make, String model, int year) {
        this.make = make;
        this.model = model;
        this.year = year;
    }

    public String getMake() {
        return make;
    }

    public void setMake(String make) {
        this.make = make;
    }

    public String getModel() {
        return model;
    }

    public void setModel(String model) {
        this.model = model;
    }

    public int getYear() {
        return year;
    }

    public void setYear(int year) {
        this.year = year;
    }
}

File: examples/lh-server/go/21-structdefs/car.go
package main

import "github.com/littlehorse-enterprises/littlehorse/sdk-go/littlehorse"

const CarStructDefName = "car"
const CarStructDefDescription = "A car."

type Car struct {
	Make  string `json:"make"`
	Model string `json:"model"`
	Year  int    `json:"year"`
}

func (Car) LHStructDef() littlehorse.LHStructDefInfo {
	return littlehorse.LHStructDefInfo{
		Name:        CarStructDefName,
		Description: CarStructDefDescription,
	}
}
